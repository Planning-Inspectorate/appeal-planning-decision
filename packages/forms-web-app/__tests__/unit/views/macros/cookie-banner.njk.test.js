const nunjucksTestRenderer = require('../nunjucks-render-helper');

describe('views/macros/cookie-banner', () => {
  it(`should have sensible default attributes`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner.njk' import cookieBanner with context -%}
    {{- cookieBanner() -}}
    `)
    ).toMatchSnapshot();
  });

  it(`should allow setting a custom serviceName`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner.njk' import cookieBanner with context -%}
    {{- cookieBanner({ serviceName: "test service name here" }) -}}
    `)
    ).toMatchSnapshot();
  });

  it(`should allow setting a custom cookiePagePath`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner.njk' import cookieBanner with context -%}
    {{- cookieBanner({ cookiePagePath: "/some/path/here" }) -}}
    `)
    ).toMatchSnapshot();
  });

  it(`should allow setting all custom attributes`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner.njk' import cookieBanner with context -%}
    {{- cookieBanner({ cookiePagePath: "/a/b/c", serviceName: "custom service name" }) -}}
    `)
    ).toMatchSnapshot();
  });
});
