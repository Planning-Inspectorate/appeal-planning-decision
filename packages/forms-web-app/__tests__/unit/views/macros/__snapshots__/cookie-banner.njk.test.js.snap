// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/macros/cookie-banner should allow setting a custom cookiePagePath 1`] = `
"
  

  

  

  

  <div class=\\"govuk-cookie-banner \\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on appeal a householder planning decision service\\" data-cy=\\"cookie-banner\\" id=\\"cookie-banner-consent\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        
        <h2 class=\\"govuk-cookie-banner__heading govuk-heading-m\\">Cookies on appeal a householder planning decision service</h2>
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>We use some essential cookies to make this service work.</p>
    <p>We'd also like to use analytics cookies so we can understand how you use the service and make improvements.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"accept\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accept-analytics-cookies\\">
              Accept analytics cookies
            </button>
          
        
          
            <button value=\\"reject\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-reject-analytics-cookies\\">
              Reject analytics cookies
            </button>
          
        
          
            
              
              
              <a class=\\"govuk-link\\" href=\\"/some/path/here\\" data-cy=\\"cookie-banner-view-cookies\\">View cookies</a>
            
          
        
      </div>
      
    </div>
  
</div>
"
`;

exports[`views/macros/cookie-banner should allow setting a custom serviceName 1`] = `
"
  

  

  

  

  <div class=\\"govuk-cookie-banner \\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on test service name here\\" data-cy=\\"cookie-banner\\" id=\\"cookie-banner-consent\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        
        <h2 class=\\"govuk-cookie-banner__heading govuk-heading-m\\">Cookies on test service name here</h2>
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>We use some essential cookies to make this service work.</p>
    <p>We'd also like to use analytics cookies so we can understand how you use the service and make improvements.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"accept\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accept-analytics-cookies\\">
              Accept analytics cookies
            </button>
          
        
          
            <button value=\\"reject\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-reject-analytics-cookies\\">
              Reject analytics cookies
            </button>
          
        
          
            
              
              
              <a class=\\"govuk-link\\" href=\\"/cookies\\" data-cy=\\"cookie-banner-view-cookies\\">View cookies</a>
            
          
        
      </div>
      
    </div>
  
</div>
"
`;

exports[`views/macros/cookie-banner should allow setting all custom attributes 1`] = `
"
  

  

  

  

  <div class=\\"govuk-cookie-banner \\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on custom service name\\" data-cy=\\"cookie-banner\\" id=\\"cookie-banner-consent\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        
        <h2 class=\\"govuk-cookie-banner__heading govuk-heading-m\\">Cookies on custom service name</h2>
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>We use some essential cookies to make this service work.</p>
    <p>We'd also like to use analytics cookies so we can understand how you use the service and make improvements.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"accept\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accept-analytics-cookies\\">
              Accept analytics cookies
            </button>
          
        
          
            <button value=\\"reject\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-reject-analytics-cookies\\">
              Reject analytics cookies
            </button>
          
        
          
            
              
              
              <a class=\\"govuk-link\\" href=\\"/a/b/c\\" data-cy=\\"cookie-banner-view-cookies\\">View cookies</a>
            
          
        
      </div>
      
    </div>
  
</div>
"
`;

exports[`views/macros/cookie-banner should have sensible default attributes 1`] = `
"
  

  

  

  

  <div class=\\"govuk-cookie-banner \\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on appeal a householder planning decision service\\" data-cy=\\"cookie-banner\\" id=\\"cookie-banner-consent\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        
        <h2 class=\\"govuk-cookie-banner__heading govuk-heading-m\\">Cookies on appeal a householder planning decision service</h2>
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>We use some essential cookies to make this service work.</p>
    <p>We'd also like to use analytics cookies so we can understand how you use the service and make improvements.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"accept\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accept-analytics-cookies\\">
              Accept analytics cookies
            </button>
          
        
          
            <button value=\\"reject\\" type=\\"submit\\" name=\\"cookie_banner\\" class=\\"govuk-button govuk-!-display-none\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-reject-analytics-cookies\\">
              Reject analytics cookies
            </button>
          
        
          
            
              
              
              <a class=\\"govuk-link\\" href=\\"/cookies\\" data-cy=\\"cookie-banner-view-cookies\\">View cookies</a>
            
          
        
      </div>
      
    </div>
  
</div>
"
`;
